src/main/java/edu/cornell/kfs/paymentworks/service/impl/PaymentWorksKfsServiceImpl.java

```diff
@@ -64,387 +59,312 @@
+			
+			if (!inactivatePayeeAchAccount(payeeAchAccount)) {
+				isSuccessful = false;
+				break;
 			}
 
-			PayeeACHAccount payeeAchAccountNew = new PayeeACHAccount();
-
-			// check if no existing ach account, and dont' have both fields
-			// coming in (can't create a new record)
-			if (ObjectUtils.isNull(payeeAchAccount)
-					&& (StringUtils.isEmpty(routingNumber) || StringUtils.isEmpty(accountNumber))) {
-				GlobalVariables.getMessageMap().putError("payeeAchAccount", KFSKeyConstants.ERROR_CUSTOM,
-						"No existing Payee ACH Account, routing number and account number required");
+			if(!validatePayeeAchAccountExistence(payeeAchAccount, routingNumber, accountNumber)) {
 				return isSuccessful;
 			}
-
-			// if we do not have an existing payee ach account by this time,
-			// create new
-			if (ObjectUtils.isNull(payeeAchAccount)) {
-				payeeAchAccountNew = new PaymentWorksAchConversionUtil().createPayeeAchAccount(vendorUpdate,
-						vendorNumber);
-			} else {
-				// if we do have a payee ach account, deep copy, and copy over
-				// the fields
-				payeeAchAccountNew = new PaymentWorksAchConversionUtil().createPayeeAchAccount(payeeAchAccount,
-						routingNumber, accountNumber);
+			
+			PayeeACHAccount payeeAchAccountNew = buildPayeeAchAccountNew(vendorUpdate, routingNumber, accountNumber, vendorNumber, payeeAchAccount);
+			if(!processPayeeAchAccountNew(payeeAchAccountNew, payeeAchAccount)) {
+				break;
 			}
+		}
 
-			// save payee ach account
+		return isSuccessful;
+	}
+	
+	protected boolean inactivatePayeeAchAccount(PayeeACHAccount payeeAchAccount) {
+		if (ObjectUtils.isNotNull(payeeAchAccount)) {
+			payeeAchAccount.setActive(false);
 			try {
-				payeeAchAccountNew = businessObjectService.save(payeeAchAccountNew);
+				payeeAchAccount = businessObjectService.save(payeeAchAccount);
 			} catch (Exception e) {
-				GlobalVariables.getMessageMap().putError("payeeAchAccount", KFSKeyConstants.ERROR_CUSTOM,
-						e.getCause().getMessage());
-
-				// roll back
-				if (ObjectUtils.isNotNull(payeeAchAccount)) {
-					payeeAchAccount.setActive(true);
-					payeeAchAccount = businessObjectService.save(payeeAchAccount);
-				}
-				isSuccessful = false;
-				break;
+				GlobalVariables.getMessageMap().putError("payeeAchAccount", KFSKeyConstants.ERROR_CUSTOM, e.getCause().getMessage());
+				return false;
 			}
-			isSuccessful = true;
 		}
-
+		return true;
+	}
+	
+	protected boolean processPayeeAchAccountNew(PayeeACHAccount payeeAchAccountNew, PayeeACHAccount payeeAchAccount) {
+		 boolean isSuccessful = true;
+		try {
+			payeeAchAccountNew = businessObjectService.save(payeeAchAccountNew);
+		} catch (Exception e) {
+			GlobalVariables.getMessageMap().putError("payeeAchAccount", KFSKeyConstants.ERROR_CUSTOM, e.getCause().getMessage());
+			if (ObjectUtils.isNotNull(payeeAchAccount)) {
```

Comment:
This is functionality that will get turned on in future user story, maybe we should weasel that logic out then?